### PR TITLE
Feat: Add spec and parser for 'nginx_log'

### PR DESCRIPTION
**Pull request by rahulXs**
Signed-off-by: Rahul Sharma <rahushar@redhat.com>

### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Complete Description of Additions/Changes:

Add new spec and parser to parse contents of nginx log file `/var/log/nginx/error.log`

### CODE REVIEW

File: insights/parsers/nginx_log.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        2022/04/02 17:59:29 [warn] 1594#1594: *711881 an upstream response is buffered to a temporary file /var/lib/nginx/tmp/uwsgi/6/25/0000003256 while reading upstream, client: 10.245.136.148, server: _, request: "GET /api/v2/labels/?page_size=200 HTTP/1.1", upstream: "uwsgi://unix:/var/run/tower/uwsgi.sock:", host: "towergtd.desjardins.com", referrer: "https://towergtd.desjardins.com/"
+        2022/04/03 00:02:28 [warn] 1591#1591: *719136 an upstream response is buffered to a temporary file /var/lib/nginx/tmp/uwsgi/7/25/0000003257 while reading upstream, client: 10.245.136.148, server: _, request: "GET /api/v2/organizations/?page_size=999 HTTP/1.1", upstream: "uwsgi://unix:/var/run/tower/uwsgi.sock:", host: "towergtd.desjardins.com"
+
+    Each line is parsed into a dictionary with the following keys:
+
+        * **raw_message(str)** - complete log line
```
Comment (psachin) [inline insights/parsers/nginx_log.py:28]: For the description can you please refer https://github.com/RedHatInsights/insights-core/blob/master/insights/parsers/openvswitch_logs.py#L17

Comment (psachin) [reply insights/parsers/nginx_log.py:28]: How about including a note in the docstring?

```
    .. note::
        Please refer to its super-class :class:`insights.core.LogFileOutput` for
        more usage information.
```

**Review by psachin — APPROVED** [review]

**Comment by huali027** [comment]
@psachin, if there is no problem with the MR, can u please merge it first? then @rahulXs can add the filter in the dummy rule. In this way, the parser and the error filter can be released in the same week, and he won't need to wait another week to check hits on RAT.

**Review by xiangce — APPROVED** [review]
Approved pending the approval of SPEC change to the assets repo.

**Comment by xiangce** [comment]
@huali027 - This PR can only be merged after the SPEC change is submitted to the assets repo and is approved too.

@rahulXs  - Since this PR requires adding a new Spec to the `uploader.json` and getting approval, please submit an MR to the assets report for this change.

Thanks.

**Comment by rahulXs** [comment]
@huali027 @xiangce Ack. I'll submit an MR to the assets repo.

**Pull request merged**